ui/effects/use-resolve-pins.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+import React from 'react';
+import useFetched from 'effects/use-fetched';
+
+type Props = {
+  pins?: { urls?: Array<string>, claimIds?: Array<string>, onlyPinForOrder?: string },
+  claimsById: { [string]: Claim },
+  doResolveClaimIds: (Array<string>) => Promise<any>,
+  doResolveUris: (Array<string>, boolean) => Promise<any>,
+};
+
+export default function useResolvePins(props: Props) {
+  const { pins, claimsById, doResolveClaimIds, doResolveUris } = props;
+
+  const [resolvedPinUris, setResolvedPinUris] = React.useState(pins ? undefined : null);
+  const [resolvingPinUris, setResolvingPinUris] = React.useState(false);
+  const hasResolvedPinUris = useFetched(resolvingPinUris);
+
+  React.useEffect(() => {
+    if (resolvedPinUris === undefined && pins && !resolvingPinUris) {
+      if (pins.urls) {
+        doResolveUris(pins.urls, true).finally(() => setResolvedPinUris(pins.urls));
+      } else if (pins.claimIds) {
+        // setResolvingPinUris is only needed for claim_search.
+        // doResolveUris uses selectResolvingUris internally to prevent double call.
+        setResolvingPinUris(true);
```

Comment:
could use this here as a `ref` maybe? since it will only be checked again in case of a double call? just nitpicking but everything looks good

Reply:
Seems like it needs to remain a state, otherwise the ref change doesn't spark a render and `const hasResolvedPinUris = useFetched(resolvingPinUris);` could be missed.